### PR TITLE
[exporter/datadog] Send host metadata on change, refactor main host metadata sending

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- [\*/datadog] Bump opentelemetry-mapping-go/\* to v0.7.1
- [exporter/datadog] Use reporter for exporter host as well

### CODE REVIEW

File: exporter/datadogexporter/internal/hostmetadata/metadata.go [hunk 1]
```diff
@@ -169,23 +168,18 @@ func RunPusher(ctx context.Context, params exporter.CreateSettings, pcfg PusherC
 		hostMetadata = metadataFromAttributes(attrs)
 	}
 	fillHostMetadata(params, pcfg, p, &hostMetadata)
-
-	// Run one first time at startup
-	if err := pusher.Push(ctx, hostMetadata); err != nil {
-		params.Logger.Warn("Initial host metadata failed", zap.Error(err))
-	} else {
-		params.Logger.Info("Sent initial host metadata")
```
Comment (songy23) [inline exporter/datadogexporter/internal/hostmetadata/metadata.go:177]: Do you want to keep this log?

File: exporter/datadogexporter/internal/hostmetadata/metadata.go [hunk 2]
```diff
@@ -169,23 +168,18 @@ func RunPusher(ctx context.Context, params exporter.CreateSettings, pcfg PusherC
...
 
 	for {
 		select {
 		case <-ctx.Done():
 			return
-		case <-ticker.C: // Send host metadata
-			if err := pusher.Push(ctx, hostMetadata); err != nil {
-				params.Logger.Warn("Sending host metadata failed", zap.Error(err))
-			} else {
-				params.Logger.Info("Sent host metadata")
```
Comment (songy23) [inline exporter/datadogexporter/internal/hostmetadata/metadata.go:188]: Same

**Review by songy23 — APPROVED** [review]
Minor comments, otherwise LGTM

**Review by mackjmr — APPROVED** [review]

File: exporter/datadogexporter/internal/hostmetadata/metadata.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/internal/hostmetadata/metadata.go:177]: We now have a log each time we push here: https://github.com/DataDog/opentelemetry-mapping-go/blob/8cccaf6f3cfb6e4ad31da562517c3a656f1692d7/pkg/inframetadata/reporter.go#L129-L132

File: exporter/datadogexporter/internal/hostmetadata/metadata.go (hunk 2 above)
Comment (mx-psi) [reply exporter/datadogexporter/internal/hostmetadata/metadata.go:188]: Ditto https://github.com/DataDog/opentelemetry-mapping-go/blob/8cccaf6f3cfb6e4ad31da562517c3a656f1692d7/pkg/inframetadata/reporter.go#L129-L132

**Pull request merged**